test/production/app-dir-edge-runtime-with-wasm/index.test.ts

```diff
@@ -0,0 +1,62 @@
+import path from 'path'
+import { createNext, FileRef } from 'e2e-utils'
+import { NextInstance } from 'test/lib/next-modes/base'
+import { renderViaHTTP } from 'next-test-utils'
+
+const files = {
+  'app/layout.tsx': `
+    export default function AppLayout({ children }) {
+      return (
+        <html>
+          <head>
+            <title>WASM Import</title>
+          </head>
+          <body>
+            {children}
+          </body>
+        </html>
+      )
+    }
+  `,
+  'app/page.tsx': `
+    // @ts-expect-error
+    import wasm from '../wasm/add.wasm?module'
+
+    console.log(wasm)
+
+    export default function Page() {
+      return "index page"
+    }
+
+    export const runtime = "experimental-edge"
+  `,
+  'wasm/add.wasm': new FileRef(path.join(__dirname, 'add.wasm')),
+}
+
+describe('app-dir edge runtime with wasm', () => {
+  let next: NextInstance
+
+  beforeAll(async () => {
+    next = await createNext({
+      files,
+      dependencies: {
+        react: 'experimental',
+        'react-dom': 'experimental',
+        typescript: 'latest',
+        '@types/react': 'latest',
+        '@types/node': 'latest',
```

Comment:
**nit:** consider writing the tests in JavaScript and not in TypeScript, so we can remove these dependencies. wdyt? <picture data-single-emoji=":eyeses:" title=":eyeses:"><img class="emoji" width="20" height="auto" src="https://single-emoji.vercel.app/api/emoji/eyJhbGciOiJkaXIiLCJlbmMiOiJBMjU2R0NNIn0..yHWGyVDgwavWe5IK.81oWO7IZtr3t2oBgugXvjyDBATPQ_eD0p4CJ2zqxiky370rRGtVSSf-T43byjjq-hXIMc236inKIXOoS-NxhFfooZJ9o9vVNfXuDR-gJGYJG64ljYOmg4g.ZJ-QVBO8-trgeDCuA7FzUg" alt=":eyeses:" align="absmiddle"></picture>

Reply:
Great idea, updated!